src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -989,14 +990,14 @@ private void startSlave(SlaveComputer computer, final TaskListener listener, Str
             computer.setChannel(session.getStdout(), session.getStdin(), listener.getLogger(), null);
         } catch (InterruptedException e) {
             session.close();
-            throw new IOException2(Messages.SSHLauncher_AbortedDuringConnectionOpen(), e);
+            throw new IOException(Messages.SSHLauncher_AbortedDuringConnectionOpen(), e);
         } catch (IOException e) {
             try {
                 // often times error this early means the JVM has died, so let's see if we can capture all stderr
                 // and exit code
-                throw new IOException2(getSessionOutcomeMessage(session,false),e);
+                throw new AbortException(getSessionOutcomeMessage(session,false));
```

Comment:
There is no reason to obscure the the friendly cause message with irrelevant stacktrace.